src/api/controller/queries.js

```diff
@@ -121,19 +121,55 @@ async function insertUser(username, email, password_hash) {
   }
 }
 
+async function getUserIdByEmail(email) {
+  const sql = 'SELECT user_id FROM user_info WHERE email = ?';
+
+  try {
+    const [rows] = await pool.query(sql, [email]);
+
+    if (rows.length === 0) {
+      throw new Error('User not found');
+    }
+
+    return rows[0].user_id;
+  } catch (err) {
+    console.error('Error in getUserIdByEmail:', err);
+    throw err;
+  }
+}
+
 // Function to add an attempt by the user to the DB.
-async function insertAttempt(username, characters_attempted, characters_missed, wpm) {
-  const sql = `
+async function insertAttempt(user_id, characters_attempted, characters_missed, wpm) {
+  const insertAttemptSQL = `
```

Comment:
Nit: I'd prefer to call this a query. 
```suggestion
  const query = `
```